### DISCUSSION TITLE
Devtools not working, debugging not helpful

### DISCUSSION THREAD

**Discussion by loxy** [apollographql/apollo-client-devtools#1186] [General]
Hi,

when using the latest Devtools (Devtools version: [4.3.1](https://github.com/apollographql/apollo-client-devtools/releases/tag/v4.3.1)) everything is empty (explorer, queries, mutations, cache).

After following debugging section on docs I see this:

![image](https://github.com/apollographql/apollo-client-devtools/assets/467620/fa8e67db-c2bf-4f78-ab7a-8e11c3af87fd)

Just to clarify: Does that mean it cannot connect somehow? Is this an error message? I would expect something like: "Error: Not possible to ..." as a first sentence, and not "Welcome ...". But I tried to check all those points anyway. So, my server is running on `/graphql` (via a proxy, this should be enough, right?) and I disabled authentication. CORS is not involved. But still it doesn't work. 

Any hints? What else can I check? Is it even required? So, does my server really needs to run on these special endpoint and authentication needs to be disabled? I cannot find anything mentioning that in docs...

I'm using latest Chrome on Windows with Apollo Client 3.8.9 in an Angular application (together with apollo-angular).

Best regards,
Kersten

**Comment by jerelmiller** [comment]
Hey @loxy 👋 

So sorry to hear you're having problems connecting! You're right, I'm not sure those debugging messages are super useful. The devtools looks for a local client instance on `window`, so queries/mutations/etc would depend on whether it could connect to your client instance, no so much the server itself.

That being said, would you see if you have a `window.__APOLLO_CLIENT__` variable in your browser console? We [set this](https://github.com/apollographql/apollo-client/blob/046296ac2285b20fbd43a621395fc4158cfbd073/src/core/ApolloClient.ts#L289) on the client for the devtool to connect. This depends on whether `connectToDevTools` is [set to `true`](https://github.com/apollographql/apollo-client/blob/046296ac2285b20fbd43a621395fc4158cfbd073/src/core/ApolloClient.ts#L272). If you don't explicity set `connectToDevTools` in options, it tries to infer it based on whether you're running a development build or not.

Let me know if you see `window.__APOLLO_CLIENT__` and I can help assist further if you're not seeing this!

**Comment by jerelmiller** [comment]
Hey @loxy 👋 

We've done a number of improvements to the devtools in recent versions. I'd love for you to try out the most recent version (4.7.0) and see if you're still having troubles seeing data or not. If so, I'd love to help you debug whats going on. Thanks!
